### PR TITLE
Feat: Aggregated api for complete meta

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Review by wingkwong — COMMENTED** [review]
1. `table` is missing in `tableCount`. only showing number of views.
2. `sharedPasswordProtected` is missing but found in your example and test. or did you rename to `sharedLockedCount` now?
3. `sharedTotal` count is incorrect. didn't find the logic.
```
"viewCount": {
	"formCount": 1,
	"gridCount": 1,
	"galleryCount": 0,
	"kanbanCount": 0,
	"total": 2,
	"sharedFormCount": 1,
	"sharedGridCount": 1,
	"sharedGalleryCount": 0,
	"sharedKanbanCount": 0,
	"sharedTotal": 0,
	"sharedLockedCount": 0
},
```
4. new apis should be [documented](https://docs-dev.nocodb.com/developer-resources/rest-apis).

**Comment by pranavxc** [comment]
> 1. `table` is missing in `tableCount`. only showing number of views.
> 2. `sharedPasswordProtected` is missing but found in your example and test. or did you rename to `sharedLockedCount` now?
> 3. `sharedTotal` count is incorrect. didn't find the logic.
> 
> ```
> "viewCount": {
> 	"formCount": 1,
> 	"gridCount": 1,
> 	"galleryCount": 0,
> 	"kanbanCount": 0,
> 	"total": 2,
> 	"sharedFormCount": 1,
> 	"sharedGridCount": 1,
> 	"sharedGalleryCount": 0,
> 	"sharedKanbanCount": 0,
> 	"sharedTotal": 0,
> 	"sharedLockedCount": 0
> },
> ```
> 
> 4. new apis should be [documented](https://docs-dev.nocodb.com/developer-resources/rest-apis).
> 
> Attached my res obj.
> 
> {"projectCount":3,"projects":[{"tableCount":{"view":0},"external":false,"viewCount":{"formCount":1,"gridCount":1,"galleryCount":0,"kanbanCount":0,"total":2,"sharedFormCount":1,"sharedGridCount":1,"sharedGalleryCount":0,"sharedKanbanCount":0,"sharedTotal":0,"sharedLockedCount":0},"webhookCount":0,"filterCount":0,"sortCount":0,"rowCount":[{"TotalRecords":"10757"}],"userCount":1},{"tableCount":{"view":7},"external":true,"viewCount":{"formCount":2,"gridCount":23,"galleryCount":0,"kanbanCount":0,"total":25,"sharedFormCount":2,"sharedGridCount":1,"sharedGalleryCount":0,"sharedKanbanCount":0,"sharedTotal":0,"sharedLockedCount":1},"webhookCount":0,"filterCount":0,"sortCount":0,"rowCount":[{"TotalRecords":"47283"}],"userCount":1},{"tableCount":{"view":0},"external":false,"viewCount":{"formCount":3,"gridCount":2,"galleryCount":0,"kanbanCount":0,"total":5,"sharedFormCount":3,"sharedGridCount":0,"sharedGalleryCount":0,"sharedKanbanCount":0,"sharedTotal":0,"sharedLockedCount":0},"webhookCount":0,"filterCount":0,"sortCount":0,"rowCount":[{"TotalRecords":"10757"}],"userCount":1}],"userCount":1,"sharedBaseCount":0}

Done the corrections :+1:

File: packages/nocodb/tests/unit/rest/tests/project.test.ts [hunk 1]
```diff
@@ -242,27 +244,76 @@ function projectTest() {
...
+          'galleryCount',
+          'kanbanCount',
+          'total',
+          'sharedFormCount',
+          'sharedGridCount',
+          'sharedGalleryCount',
+          'sharedKanbanCount',
+          'sharedTotal',
+          'sharedLockedCount')
+        expect(res.body.projects[0]).have.keys(
```
Comment (wingkwong) [inline packages/nocodb/tests/unit/rest/tests/project.test.ts:303]: missing `external`

File: scripts/sdk/swagger.json [hunk 2]
```diff
@@ -5337,6 +5337,165 @@
...
+                        }
+                      ],
+                      "userCount": 1,
+                      "sharedBaseCount": 0
+                    }
+                  }
+                },
+                "examples": {
+                  "example-1": {
+                    "value": "{\n  \"projectCount\": 1,\n  \"projects\": [\n    {\n      \"tableCount\": {\n        \"table\": 3,\n        \"view\": 0\n      },\n      \"viewCount\": {\n        \"formCount\": 0,\n        \"gridCount\": 3,\n        \"galleryCount\": 0,\n        \"kanbanCount\": 0,\n        \"total\": 3,\n        \"sharedFormCount\": 0,\n        \"sharedGridCount\": 0,\n        \"sharedGalleryCount\": 0,\n        \"sharedKanbanCount\": 0,\n        \"sharedTotal\": 0,\n        \"sharedLockedCount\": 0\n      },\n      \"webhookCount\": 0,\n      \"filterCount\": 0,\n      \"sortCount\": 0,\n      \"rowCount\": [\n        {\n          \"TotalRecords\": \"76\"\n        }\n      ],\n      \"userCount\": 1\n    }\n  ],\n  \"userCount\": 1,\n  \"sharedBaseCount\": 0\n}"
```
Comment (wingkwong) [inline scripts/sdk/swagger.json:5486]: may double check to see if it's outdated.

File: packages/nocodb/tests/unit/rest/tests/project.test.ts (hunk 1 above)
Comment (pranavxc) [reply packages/nocodb/tests/unit/rest/tests/project.test.ts:303]: Updated now :+1:

File: scripts/sdk/swagger.json (hunk 2 above)
Comment (pranavxc) [reply scripts/sdk/swagger.json:5660]: Updated now :+1:

**Review by wingkwong — APPROVED** [review]
wait for CI/CD.

**Pull request merged**